utils/src/treev2/quickPickWizard/experiences/subscriptionExperience.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../../../index';
+import * as vscode from 'vscode';
+import { AzureWizard } from '../../../wizard/AzureWizard';
+import { QuickPickAzureSubscriptionStep } from '../quickPickAzureResource/QuickPickAzureSubscriptionStep';
+import { ApplicationSubscription, PickSubscriptionWizardContext } from '../../../../hostapi.v2';
+import { ResourceGroupsItem } from '../quickPickAzureResource/tempTypes';
+import { NoResourceFoundError } from '../../../errors';
+
+export async function subscriptionExperience(context: types.IActionContext, tdp: vscode.TreeDataProvider<ResourceGroupsItem>): Promise<ApplicationSubscription> {
```

Comment:
Should the experience just have an ExperienceBase class that does all this boilerplate? You have to declare the wizard, wizard context, prompt every experience.

Reply:
The API I keep mentioning that I'm working on fixes this problem. It's an "experience builder" tool. You'll use it to build an experience by adding the steps you want with the settings you want. Still WIP